drivers/spi/spi_mcux_dspi.c

```diff
@@ -299,18 +863,32 @@ static const struct spi_driver_api spi_mcux_driver_api = {
 		.transfer_delay = UTIL_AND(				\
 			DT_INST_NODE_HAS_PROP(id, transfer_delay),	\
 			DT_INST_PROP(id, transfer_delay)),		\
-	};								\
-	static struct spi_mcux_data spi_mcux_data_##id = {		\
-		SPI_CONTEXT_INIT_LOCK(spi_mcux_data_##id, ctx),		\
-		SPI_CONTEXT_INIT_SYNC(spi_mcux_data_##id, ctx),		\
+		.whichCtar = UTIL_AND(					\
+			DT_INST_NODE_HAS_PROP(id, ctar),		\
+			DT_INST_PROP(id, ctar)),			\
+		.samplePoint = UTIL_AND(				\
+			DT_INST_NODE_HAS_PROP(id, sample_point),	\
+			DT_INST_PROP(id, sample_point)),		\
+		.enableContinuousSCK = UTIL_AND(			\
+			DT_INST_NODE_HAS_PROP(id, continuous_sck),	\
+			DT_INST_PROP(id, continuous_sck)),		\
+		.enableRxFifoOverWrite = UTIL_AND(			\
+			DT_INST_NODE_HAS_PROP(id, rx_fifo_overwrite),	\
+			DT_INST_PROP(id, rx_fifo_overwrite)),		\
+		.enableModifiedTimingFormat = UTIL_AND(			\
+		    DT_INST_NODE_HAS_PROP(id, modified_timing_format),	\
+		    DT_INST_PROP(id, modified_timing_format)),		\
+		.is_dma_chn_shared = UTIL_AND(				\
+		    DT_INST_NODE_HAS_PROP(id, nxp_rx_tx_chn_share),	\
+		    DT_INST_PROP(id, nxp_rx_tx_chn_share)),		\
```

Comment:
For the boolean properties, you can just use `DT_INST_PROP`, you don't need to check for existence. For the other properties, would `DT_INST_PROP_OR` work better?

Reply:
updated

Reply:
updated